### PR TITLE
dnsdist: Add unit tests for the Lua FFI interface

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

<strike>This PR is based on #12417 and will have to be rebased.</strike>

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [x] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdistdist/test-dnsdist-lua-ffi.cc [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+    BOOST_CHECK_EQUAL(dnsdist_ffi_dnsquestion_get_local_port(&lightDQ), 53U);
+  }
+
+  {
+    // dnsdist_ffi_dnsquestion_get_remoteaddr, dnsdist_ffi_dnsquestion_get_remote_port
+    const char* buffer = nullptr;
+    size_t bufferSize = 0;
+    dnsdist_ffi_dnsquestion_get_remoteaddr(&lightDQ, reinterpret_cast<const void**>(&buffer), &bufferSize);
+    BOOST_REQUIRE(buffer != nullptr);
+    BOOST_REQUIRE_EQUAL(bufferSize, sizeof(ids.origDest.sin4.sin_addr.s_addr));
```
Comment (chbruyand) [inline pdns/dnsdistdist/test-dnsdist-lua-ffi.cc:85]:
```suggestion
    BOOST_REQUIRE_EQUAL(bufferSize, sizeof(ids.origRemote.sin4.sin_addr.s_addr));
```

File: pdns/dnsdistdist/test-dnsdist-lua-ffi.cc [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
+  {
+    // test V6 as well
+    ids.origRemote = ComboAddress("[2001:db8::1]:65535");
+    ids.origDest = ComboAddress("[2001:db8::2]:53");
+
+    const char* buffer = nullptr;
+    size_t bufferSize = 0;
+    dnsdist_ffi_dnsquestion_get_remoteaddr(&lightDQ, reinterpret_cast<const void**>(&buffer), &bufferSize);
+    BOOST_REQUIRE(buffer != nullptr);
+    BOOST_REQUIRE_EQUAL(bufferSize, sizeof(ids.origDest.sin6.sin6_addr.s6_addr));
```
Comment (chbruyand) [inline pdns/dnsdistdist/test-dnsdist-lua-ffi.cc:133]:
```suggestion
    BOOST_REQUIRE_EQUAL(bufferSize, sizeof(ids.origRemote.sin6.sin6_addr.s6_addr));
```

**Review by chbruyand — APPROVED** [review]
lgtm (although I think there may be two small nits)

**Comment by rgacogne** [comment]
> lgtm (although I think there may be two small nits)

Good catch, I just pushed a fix implementing your suggestion, thanks!

**Pull request merged**